src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -167,6 +192,16 @@ private[impl] class InstanceTrackerActor(
     }
   }
 
+  private def updateRepository(repositoryFunc: () => Future[Done], ack: Ack)(implicit ec: ExecutionContext): Future[RepositoryStateUpdated] = {
+    repositoryFunc()
+      .recoverWith(tryToRecoverRepositoryFailure(ack.effect))
+      .map(_ => RepositoryStateUpdated(ack))
+      .recoverWith {
+        // if we could not recover from repository failure, propagate the error
+        case NonFatal(e) => Future.successful(RepositoryStateUpdated(Ack(ack.initiator, InstanceUpdateEffect.Failure(e))))
```

Comment:
In case it's a `Fatal` exception, pipeTo will [wrap](https://github.com/akka/akka/blob/v2.5.7/akka-actor/src/main/scala/akka/pattern/PipeToSupport.scala#L20) it into `Status.Failure(f)` and send as a message to this actor, correct? I think you need to handle this message with an extra `Receive` case.

Reply:
good catch! Thanks ... I think I just want to throw the exception in that case... (fixed)